reactor-core/src/main/java/reactor/core/publisher/MonoZip.java

```diff
@@ -333,13 +339,26 @@ public void onNext(Object t) {
 		@Override
 		public void onError(Throwable t) {
 			error = t;
-			parent.signalError(t);
+			if (parent.delayError) {
+				parent.signal();
+			}
+			else {
```

Comment:
shouldn't this be simplified to `parent.signalError(this, t)` ? otherwise the `signalError` method is only used in test!